### ISSUE TITLE
Graph Embedding 

### ISSUE THREAD

**Issue by itsshavar** [networkx/networkx#4056]
Hi,
Any plans on implementation of embedding or vector representation of nodes and edges. @dschult

**Comment by dschult** [comment]
Can you be more specific?

**Comment by itsshavar** [comment]
I am talking about if we can represent a node using some dimension. 
something like Node2Vec. <https://arxiv.org/abs/1607.00653>

**Comment by MridulS** [comment]
node2vec implementation from the authors of the paper https://github.com/aditya-grover/node2vec

**Comment by MridulS** [comment]
https://github.com/chihming/awesome-network-embedding is a nice resource to find  implementations of network embedding algorithms.

**Comment by itsshavar** [comment]
currently any deep learning framework is not part of networkX repo so i think that is not advisable and as far as node2vec concerned we can use node2vec bcz it does not uses any deep learning framework.

**Comment by dschult** [comment]
This sounds promising.

**Comment by MridulS** [comment]
@itsshavar It doesn't depend on deep learning but node2vec builds on top of word2vec, it's still a learning algorithm.

**Comment by MridulS** [comment]
@dschult do you envision adding graph learning algorithms in networkx? I know of atleast one library which builds on top of networkx with graph learning algorithms, https://github.com/benedekrozemberczki/karateclub

**Comment by itsshavar** [comment]
@MridulS  Yes it's a learning algorithm with basic weight adjustment and skip gram method. And as far as above link is concerned, they are also using gensim to implement the same. I am talking about doing something from scratch meaning not using any library to attain the same.

**Comment by dschult** [comment]
I think the boundary between graph learning algorithms and graph algorithms is pretty porous. I think there is a fair region of overlap where algorithms of one can be included in packages for the other.  We'd like to expand connections with the graph algorithms in sklearn. The package you point to also looks pretty good -- so I think we should support these other packages and also make room for tools at the boundary (which sometimes might extend over the boundary).   How's that for a wishy-washy answer.  A lot depends on interests and time of developers too of course.

**Comment by rossbar** [comment]
This one's been stale for a while, so I will go ahead and close it.

I think @dschult 's take in the previous comment summarizes the discussion well. Note also in the meantime that the proliferation/advancement of graph learning libraries and related technology has exploded. Any contributors who are interested in such things are of course encouraged to start discussions/propose ideas in the NX repo!
